packages/neptune-extension/src/common/state/notebook/reducer.ts

```diff
@@ -1,24 +1,78 @@
-import { Reducer } from 'redux';
 import { NotebookActions } from './actions'
 import { NotebookDTO } from 'generated/leaderboard-client/src/models';
 
 interface NotebookState {
   loaded: boolean
   notebook?: NotebookDTO
+  uploadNotebookStatus: 'none' | 'pending' | 'success' | 'failure'
+  uploadCheckpointStatus: 'none' | 'pending' | 'success' | 'failure'
 }
 
 const initialState: NotebookState = {
   loaded: false,
```

Comment:
This flag was originally used to signal that initial notebook loading has been resolved (or did not happen at all). The name may be misleading, it should have been `initialized` probably. Nevertheless it cannot be reused for modal loaded flag.